encodings/fastlanes/src/bitpacking/compute/search_sorted.rs

```diff
@@ -0,0 +1,94 @@
+use std::cmp::Ordering;
+
+use fastlanes::BitPacking;
+use vortex::compute::search_sorted::{
+    search_sorted, IndexOrd, Len, SearchResult, SearchSorted, SearchSortedFn, SearchSortedSide,
+};
+use vortex::ArrayDType;
+use vortex_dtype::{match_each_unsigned_integer_ptype, NativePType};
+use vortex_error::VortexResult;
+use vortex_scalar::Scalar;
+
+use crate::{unpack_single_primitive, BitPackedArray};
+
+impl SearchSortedFn for BitPackedArray {
+    fn search_sorted(&self, value: &Scalar, side: SearchSortedSide) -> VortexResult<SearchResult> {
+        let ptype = self.ptype();
+        match_each_unsigned_integer_ptype!(ptype, |$P| {
+            let unwrapped_value: $P = value.cast(self.dtype())?.try_into().unwrap();
+            if let Some(patches_array) = self.patches() {
+                if (unwrapped_value.leading_zeros() as usize) < ptype.bit_width() - self.bit_width() {
+                    search_sorted(&patches_array, value.clone(), side)
+                } else {
+                    Ok(SearchSorted::search_sorted(&BitPackedSearch(self), &unwrapped_value, side))
```

Comment:
I don't think this works? If you binary search over a patched array, you have no guarantee what the values are underneath the patch, so it may not be sorted and all your assumptions go out the window

Reply:
I think as long as patches are less than 50% of the array it will work. Anyway can add some safety to it

Reply:
Hmm, maybe a safer thing here, because this only works on sorted arrays, is to say whether the search value is less than the minimum patch?

Reply:
Changed the logic to immediately bail if we try to read index that's patched